### PR TITLE
Improvements to 7.0 compat

### PR DESCRIPTION
**Pull request by craigh**
| Q                 | A
| ----------------- | ---
| Bug fix?          | no
| New feature?      | no
| BC breaks?        | no
| Deprecations?     | no
| Fixed tickets     | -
| Refs tickets      | -
| License           | MIT
| Changelog updated | no

## Description
updates for improved compatibility with PHP 7 and minor improvements to avoid Session usage when using CLI.

### CODE REVIEW

File: src/lib/Zikula/Bundle/CoreInstallerBundle/Controller/AjaxUpgradeController.php [hunk 1]
```diff
@@ -188,9 +188,11 @@ private function from141to142()
             return true;
         }
         // do some clean up
-        \SessionUtil::delVar('interactive_init');
-        \SessionUtil::delVar('interactive_remove');
-        \SessionUtil::delVar('interactive_upgrade');
+        if ($this->container->get('request_stack')->getCurrentRequest() && $this->container->get('request_stack')->getCurrentRequest()->hasSession()) {
```
Comment (cmfcmf) [inline src/lib/Zikula/Bundle/CoreInstallerBundle/Controller/AjaxUpgradeController.php]: Uuhm. Maybe introduce a variable for the current request? :boat:

File: src/system/RoutesModule/Twig/TwigExtension.php [hunk 2]
```diff
@@ -135,4 +135,32 @@ public function displayPathAsString($path, $route)
...
+    {
+    }
+
+    public function getImageThumb($params)
+    {
+    }
+
+    public function profileLink($uid, $class = '', $maxLength = 0)
+    {
+    }
```
Comment (cmfcmf) [inline src/system/RoutesModule/Twig/TwigExtension.php]: Why are these needed :question:

Comment (craigh) [inline src/system/RoutesModule/Twig/TwigExtension.php]: they call session related functions and get instantiated on install causing the php7 build to fail

Comment (craigh) [inline src/system/RoutesModule/Twig/TwigExtension.php]: and they are generated functions that are not needed in the core.

Comment (cmfcmf) [inline src/system/RoutesModule/Twig/TwigExtension.php]: Oh I see. They overwrite the functions in the base class.

Comment (craigh) [inline src/system/RoutesModule/Twig/TwigExtension.php]: yes 😀

Comment (Guite) [inline src/system/RoutesModule/Twig/TwigExtension.php]: `getUserAvatar` is used here:
https://github.com/zikula/core/blob/1.4/src/system/RoutesModule/Resources/views/Helper/includeStandardFieldsDisplay.html.twig#L18
https://github.com/zikula/core/blob/1.4/src/system/RoutesModule/Resources/views/Helper/includeStandardFieldsDisplay.html.twig#L25
So you might remove that as well.
